PowerEditor/installer/nppSetup.nsi

```diff
@@ -187,8 +187,15 @@ updaterDone:
 		StrCpy $LANGUAGE "$tempLng" ; set default language
 	${EndIf}
 	
+	Call unsupportedLanguageToEnglish
+	
 	!insertmacro MUI_LANGDLL_DISPLAY
-
+	
+	; check for the possible NSIS lang-engine fault
+	; - some users (maybe with some Windows server editions?) reported a complete omission of the installer language-selection dlg
+	;   (the Welcome Screen is then shown directly instead and the $LANGUAGE var is filled up with the "Installer Language" nonsense)
+	Call unsupportedLanguageToEnglish

```

Comment:
And why `unsupportedLanguageToEnglish` is called again here?

Reply:
Because otherwise the NSIS can left us here with $LANGUAGE=="Installer Language" and then after .onInit any [existing](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/591b00e538052543376e937a220aa6ef024ed10b/PowerEditor/installer/nsisInclude/mainSectionFuncs.nsh#L129) (or future) code dependent on the $LANGUAGE will have a problem...

Reply:
I see your point.
Why the first call then?

Reply:
Hmm, it's getting really complicated. Short answer is :

Now I think the 1st call is no longer necessary because of that other patch (English-1st-in-our-lang-list).
But there is also no harm in left it there.

Explanation:
First such call had to be there for any user with the NSIS lang-engine working without inherent failures, e.g.:
- user has Icelandic Windows locale
- on the installer start, before our .onInit had been called, the $LANGUAGE variable was preset by NSIS with its Icelandic equivalent value
- then this was offered as default for the language selection dlg combobox
- because we do not have NSIS-lang for that Icelandic, Afrikaans had been correctly (it was our 1st choice before) preselected instead in the lang-combobox by NSIS
- but now we switched to the English-1st, so that 1st call should not be needed anymore there (sorry, I did not really tested, no more time...)

Reply:
@xomx 
Here the current code in master:

```nsis
	; look for previously selected language
	ClearErrors
	Var /GLOBAL tempLng
	ReadRegStr $tempLng HKLM "SOFTWARE\${APPNAME}" 'InstallerLanguage'
	${IfNot} ${Errors}
		StrCpy $LANGUAGE "$tempLng" ; set default language
	${EndIf}
	
	!insertmacro MUI_LANGDLL_DISPLAY

	; save selected language to registry
	WriteRegStr HKLM "SOFTWARE\${APPNAME}" 'InstallerLanguage' '$Language'
```

The code does the following stuffs:
1. read a value from registry, fills $Language with the value.
2. then NSIS macro MUI_LANGDLL_DISPLAY uses $Language to pre-select a language for user. And $Language can be changed, If user changes the selection. 
3. write the value of $Language into registry for the next installation.


With only your fix **English-1st-in-our-lang-list**, in case of problem of read/write value from/into registry, $Language will be always the value we expected (1033). So it's the fix sufficient and there won't be the problem of copying nativeLang.xml file.

> But there is also no harm in left it there.

The PR #14125 of @mpheath should work (without the fix **English-1st-in-our-lang-list**) but for unknown reason it doesn't. So it's reasonable IMO to not add this code on which we know it doesn't work.

Please keep only **English-1st-in-our-lang-list** and remove other unnecessary code.

Reply:
@xomx Recognition of the issue as being valid has helped to propose a fix (or two if needed). I agree with a minimal fix as suggested by @donho and then evaluate again after a release to see if the issue has been fixed or needs more attention.

Reply:
@donho 

> Please keep only English-1st-in-our-lang-list and remove other unnecessary code.

Done in #14224 .
Sorry for the delay (no time and I am on the GH web-interface only).

> With only your fix English-1st-in-our-lang-list, in case of problem of read/write value from/into registry, $Language will be always the value we expected (1033).

No. I obviously failed here as a explainer :-)

Reply:
@xomx 
> Done in https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14224 .

Thank you!

> No. I obviously failed here as an explainer :-)

I may miss somethings...
But when I am checking the code, I don't see how $Language is not 1033 if we apply English-1st-in-our-lang-list fix.